### PR TITLE
fix(bq2bq): handle BQ dry run error caused by self table SQL reference

### PR DESCRIPTION
**Pull request by vianhazman**
Closes https://github.com/odpf/transformers/issues/4

### CODE REVIEW

File: task/bq2bq/main.go [hunk 1]
```diff
@@ -389,16 +389,27 @@ func (b *BQ2BQ) GenerateDependencies(ctx context.Context, request models.Generat
...
+	parsedDependencies, ignoredDependencies, err := b.FindDependenciesWithRegex(ctx, request, selfTable.Destination)
 	if err != nil {
 		return response, err
 	}
 
 	// try to resolve referenced tables directly from BQ APIs
 	response.Dependencies, err = b.FindDependenciesWithRetryableDryRun(timeoutCtx, queryData.Value, svcAcc)
 	if err != nil {
-		return response, err
+		if !strings.Contains(err.Error(), fmt.Sprintf("Not found: Table %s was not found", selfTable.Destination)) {
```
Comment (kushsharma) [inline task/bq2bq/main.go:410]: can you add a comment here as well stating why we are doing this?

**Review by kushsharma — CHANGES_REQUESTED** [review]
Thank you for the PR @vianhazman LGTM, just some minor updates.

File: task/bq2bq/main.go [hunk 2]
```diff
@@ -486,7 +488,7 @@ func (b *BQ2BQ) GenerateDependencies(ctx context.Context, request models.Generat
 // they're a single sequence of characters. But on the other hand
 // this also means that otherwise valid reference to "dataset.table"
 // will not be recognised.
-func (b *BQ2BQ) FindDependenciesWithRegex(ctx context.Context, request models.GenerateDependenciesRequest) ([]string, []string, error) {
+func (b *BQ2BQ) FindDependenciesWithRegex(ctx context.Context, request models.GenerateDependenciesRequest, destination string) ([]string, []string, error) {
```
[the remark this answers is not in the document]
Comment (vianhazman) [reply task/bq2bq/main.go]: I don't think so, i've refactored `FindDependenciesWithRegex` function so we can just pass the `queryString` as an param.

File: task/bq2bq/main.go (hunk 1 above)
Comment (vianhazman) [reply task/bq2bq/main.go:413]: Sure, added the comments

**Comment by kushsharma** [comment]
LGTM

**Pull request merged**